### PR TITLE
Retry failed commands and cleanup

### PR DESCRIPTION
**Pull request by zheins**
# What? Why?
eVOLVER server now broadcasts failed commands back via websocket. This PR allows the DPU to log and act on that information, retrying failed commands.

Changes proposed in this pull request:
- Keep track of commands from the last broadcast
- If a failed command message is received, check if the failed command is one that the DPU recently sent. If it is, retry it (up to 3 times)
- Some cleanup of class vars, `__init()__` function

### CODE REVIEW

File: experiment/template/eVOLVER.py [hunk 1]
```diff
@@ -126,6 +135,13 @@ def on_activecalibrations(self, data):
                                 self._create_file(x, param + '_raw', defaults=[exp_str])
                     break
 
+    def on_serialexception(self, data):
+        logger.error('Failed command!')
+        logger.error(data)
+        if data in self.commands and retry_count <= RETRY_LIMIT:
+            self.emit('command', data, namespace = '/dpu-evolver')
+            retry_count = retry_count + 1
```
Comment (mgalardini) [inline experiment/template/eVOLVER.py]: Should be `self.retry_count += 1`?

File: experiment/template/eVOLVER.py [hunk 2]
```diff
@@ -126,6 +135,13 @@ def on_activecalibrations(self, data):
                                 self._create_file(x, param + '_raw', defaults=[exp_str])
                     break
 
+    def on_serialexception(self, data):
+        logger.error('Failed command!')
+        logger.error(data)
+        if data in self.commands and retry_count <= RETRY_LIMIT:
```
Comment (mgalardini) [inline experiment/template/eVOLVER.py]: should be `self.retry_count`?

Comment (zheins) [reply experiment/template/eVOLVER.py]: yes, thanks!

File: experiment/template/eVOLVER.py (hunk 1 above)
Comment (zheins) [reply experiment/template/eVOLVER.py]: I've been switching between languages too much lately, can't remember how to increment variables anymore :joy:

**Comment by zheins** [comment]
Out of date now. The core parts were fixed elsewhere - will revisit at a later date.

**Pull request closed without merging**